crates/project_panel/src/project_panel.rs

```diff
@@ -887,6 +890,8 @@ impl ProjectPanel {
             let should_hide_rename = is_root
                 && (cfg!(target_os = "windows")
                     || (settings.hide_root && visible_worktrees_count == 1));
+            let selected_files_count = self.collect_selected_files(cx).len();
+            let should_show_compare = !is_dir && selected_files_count >= 2;
```

Comment:
(would need to remove the above line too)

```suggestion
            let should_show_compare = !is_dir && self.collect_selected_files(cx).len() > 1;
```

This way, Rust would short-circuit and won't execute the right expression if we're in the directory, this skipping the allocations.

This is a rendering code, being called 120 times/second sometimes, so might be worth rewriting this new code into something that never allocates.